compiler/src/dotty/tools/dotc/ast/TreeInfo.scala

```diff
@@ -426,8 +426,17 @@ trait TypedTreeInfo extends TreeInfo[Type] { self: Trees.Instance[Type] =>
    */
   def constToLiteral(tree: Tree)(implicit ctx: Context): Tree = {
     val tree1 = ConstFold(tree)
+    def canInlineConstant(value: Constant): Boolean = {
+      isIdempotentExpr(tree1) && // see note in documentation
+      // lazy value must be initialized (would not be needed with isPureExpr)
+      !tree1.symbol.is(Lazy) &&
+      // could hide initialization order issues (ex. val with constant type read before initialized)
+      (!ctx.owner.isLocalDummy || (tree1.symbol.is(Method) || value.isZero) ||
```

Comment:
`tree1.symbol.is(Method)`

Why is inlining the return value of a (possibly-side-effectfull) method safe?